src/hearings/converters/hearing-specific-date.answer.converter.ts

```diff
@@ -9,32 +9,30 @@ export class HearingSpecificDateAnswerConverter implements AnswerConverter {
-        } else if (hearingWindow && hearingWindow.dateRangeStart && !hearingWindow.dateRangeEnd) {
-          specificDateSelection = RadioOptions.YES;
-          const earliestHearingDate = moment(hearingWindow.dateRangeStart).format(HearingDateEnum.DisplayMonth);
-          specificDateSelection += earliestHearingDate !== HearingDateEnum.InvalidDate ? `<dt class="heading-h3 bottom-0">Earliest hearing date</dt>${earliestHearingDate}` : '';
-        } else if (hearingWindow && !hearingWindow.dateRangeStart && hearingWindow.dateRangeEnd) {
-          specificDateSelection = RadioOptions.YES;
-          const latestHearingDate = moment(hearingWindow.dateRangeEnd).format(HearingDateEnum.DisplayMonth);
-          specificDateSelection += latestHearingDate !== HearingDateEnum.InvalidDate ? `<dt class="heading-h3 bottom-0">Latest hearing date</dt>${latestHearingDate}` : '';
-        } else if (hearingWindow && hearingWindow.firstDateTimeMustBe) {
-          specificDateSelection = RadioOptions.YES;
-          const firstDate = moment(hearingWindow.firstDateTimeMustBe).format(HearingDateEnum.DisplayMonth);
-          specificDateSelection += firstDate !== HearingDateEnum.InvalidDate ? `<dt class="heading-h3 bottom-0">The first date of the hearing must be</dt>${firstDate}` : '';
-        } else if (hearingWindow) {
-          specificDateSelection = RadioOptions.NO;
-        }
-        return specificDateSelection;
-      })
+      map(state => this.createAnswer(state))
     );
   }
+
+  private createAnswer(state: State): string {
+    let specificDateSelection: string = '';
+    let earliestHearingDate: string = '';
+    let latestHearingDate: string = '';
+    const hearingWindow = state.hearingRequest.hearingRequestMainModel.hearingDetails.hearingWindow;
+
+    if (hearingWindow && (hearingWindow.dateRangeStart || hearingWindow.dateRangeEnd)) {
+      specificDateSelection = RadioOptions.CHOOSE_DATE_RANGE;
+      earliestHearingDate = moment(hearingWindow.dateRangeStart).format(HearingDateEnum.DisplayMonth);
+      latestHearingDate = moment(hearingWindow.dateRangeEnd).format(HearingDateEnum.DisplayMonth);
+      specificDateSelection += earliestHearingDate !== HearingDateEnum.InvalidDate ? `<dt class="heading-h3 bottom-0">Earliest hearing date</dt>${earliestHearingDate}` : '';
+      specificDateSelection += latestHearingDate !== HearingDateEnum.InvalidDate ? `<dt class="heading-h3 bottom-0">Latest hearing date</dt>${latestHearingDate}` : '';
+    } else if (hearingWindow && hearingWindow.firstDateTimeMustBe) {
+      specificDateSelection = RadioOptions.YES;
+      const firstDate = moment(hearingWindow.firstDateTimeMustBe).format(HearingDateEnum.DisplayMonth);
+      specificDateSelection += earliestHearingDate !== HearingDateEnum.InvalidDate ? `<dt class="heading-h3 bottom-0">The first date of the hearing must be</dt>${firstDate}` : '';
```

Comment:
assume this earliestHearingDate=>firstDate?

Reply:
done